### ISSUE TITLE
svg vs png and their performance

### ISSUE THREAD

**Issue by sobolvb** [osmdroid/osmdroid#1511]
[х] Question
[х] Performance

v. 6.1.2

Hello!
I don’t know if this applies to the bonus pack, but it seems to me that not really.
I have 3 questions.
1) After parsing kml, I use styler for each element and this greatly increases the loading time of the kml file. Is there any way to optimize this?
2) On video number 1, I use vector images for icons, but there are a lot of them and they too affect performance. Is there really a reason for the number of objects, or can I somehow optimize it?
3) I use vector ones for the reason from video number 2, there I use png format icons and they don’t scale up,they just fly away from their positions, can I fix it somehow and optimize them again so that it doesn't lag? 

https://youtu.be/MezLPLDZAHU - Video number 1
https://youtu.be/GgqIsSp3c8s - Video number 2

**Comment by mat8854** [comment]
@sobolvb Have you considered using [Marker Clustering](https://github.com/MKergall/osmbonuspack/wiki/Tutorial_3)? Easy and fast solution. It helps performance and UX wise.

**Comment by sobolvb** [comment]
@mat8854 
Yes, I tried it and productivity really increased. But the fact is that not all my kml files contain only points, basically these are lines with points placed on them, so this is not an option.

**Comment by mat8854** [comment]
@sobolvb Okay, another approach is testing enabled HW-acceleration:

`Configuration.getInstance().setMapViewHardwareAccelerated(true);`

I have an older codebase with a feature toggle for that. There are still performace issues with polylines I think.

**Comment by sobolvb** [comment]
@mat8854 
Thank you, this improved the situation, albeit very quietly. Although now I have a version with svg icons installed, I would like to find a solution for flying raster icons and try HardwareAccelerated with them

**Comment by mat8854** [comment]
@sobolvb And it it also worth to look into the Flutter map packages and addons from time to time. They have become quite mature over the last two years. It depends what you want to build. The map rendering behaves different there. It could be an alternative.
